zerver/actions/streams.py

```diff
@@ -222,7 +208,8 @@ def deactivated_streams_by_old_name(realm: Realm, stream_name: str) -> QuerySet[
         # characters, followed by `!DEACTIVATED:`, followed by at
         # most MAX_NAME_LENGTH-(length of the prefix) of the name
         # they provided:
-        Q(name__regex=rf"^{fixed_length_prefix}{truncated_name}")
+        Q(name=stream_name)
+        | Q(name__regex=rf"^{fixed_length_prefix}{truncated_name}")
```

Comment:
Should this block of code be removed, given we don't expect to have remaining streams with `DEACTIVATED` in the name? Or is this to handle streams we can't properly rename in that migration?

This is part of  the `zerver/management/commands/unarchive_stream.py` management command, and not used for anything else.

Reply:
> is this to handle streams we can't properly rename in that migration

Yes, since migration cannot guarantee the removal of the DEACTIVATED prefix from all archived streams, as many archived streams may retain their old names.